quickwit-storage/src/cache/disk/cache.rs

```diff
@@ -0,0 +1,330 @@
+use std::io;
+use std::ops::Range;
+use std::path::{Path, PathBuf};
+use std::sync::atomic::{AtomicU64, AtomicUsize, Ordering};
+use std::time::Duration;
+
+use async_trait::async_trait;
+use parking_lot::Mutex;
+use tracing::{error, warn};
+
+use super::file::FileKey;
+use super::store::{open_disk_store, FileBackedDirectory, Store};
+use crate::cache::Cache;
+use crate::metrics::CacheMetrics;
+use crate::OwnedBytes;
+
+/// TODO: Move to shared area for memory and disk cache.
+const MIN_TIME_SINCE_LAST_ACCESS: Duration = Duration::from_secs(60);
+
+type LruCache = lru::LruCache<FileKey, ()>;
+
+/// TODO: Move to shared area for memory and disk cache.
+#[derive(Clone, Copy, Debug)]
+pub(crate) enum Capacity {
+    Unlimited,
+    InBytes(usize),
+}
+
+impl Capacity {
+    fn exceeds_capacity(&self, num_bytes: usize) -> bool {
+        match *self {
+            Capacity::Unlimited => false,
+            Capacity::InBytes(capacity_in_bytes) => num_bytes > capacity_in_bytes,
+        }
+    }
+}
+
```

Comment:
Realistically this can be moved out into a shared file so that we're not duplicating code across the memory and disk implementation.